app/assets/javascripts/darkswarm/directives/enterprise_modal.js.coffee

```diff
@@ -1,9 +1,15 @@
-Darkswarm.directive "enterpriseModal", ($modal)->
+Darkswarm.directive "enterpriseModal", ($modal, Enterprises, EnterpriseResource) ->
   restrict: 'E'
   replace: true
   template: "<a ng-transclude></a>"
   transclude: true
-  link: (scope, elem, attrs, ctrl)->
-    elem.on "click", (ev)=>
+  link: (scope, elem, attrs, ctrl) ->
+    elem.on "click", (ev) =>
       ev.stopPropagation()
+      params =
+        id: scope.enterprise.id
+      EnterpriseResource.relatives params, (data) =>
+        Enterprises.addEnterprises data
+        scope.enterprise = Enterprises.enterprises_by_id[scope.enterprise.id]
+        Enterprises.dereferenceEnterprise scope.enterprise
```

Comment:
If the constructor of the Enterprises service dereferences all enterprises, should `addEnterprises` automatically dereference any new enterprises?